### PR TITLE
New Profile Picture Placeholder

### PR DESCRIPTION
**Pull request by metaphore**
This implementation replaces the old `JazzIdenticonDrawable` with `AvatarPlaceholderGenerator` that creates static bitmap avatars based on the recipient's public key and display name.

### CODE REVIEW

File: src/org/thoughtcrime/securesms/notifications/SingleRecipientNotificationBuilder.java [hunk 1]
```diff
@@ -321,4 +320,14 @@ private CharSequence getBigText(List<CharSequence> messageBodies) {
     return content;
   }
 
+  private static Drawable obtainAvatarPlaceholderDrawable(Context context, Recipient recipient) {
```
Comment (nielsandriesse) [inline src/org/thoughtcrime/securesms/notifications/SingleRecipientNotificationBuilder.java:323]: What about group notifications?

File: src/org/thoughtcrime/securesms/loki/activities/SettingsActivity.kt [hunk 2]
```diff
@@ -74,14 +74,17 @@ class SettingsActivity : PassphraseRequiredActionBarActivity() {
 
         setContentView(R.layout.activity_settings)
 
+        val origUserDisplayName = DatabaseFactory.getLokiUserDatabase(this).getDisplayName(hexEncodedPublicKey)
```
Comment (nielsandriesse) [inline src/org/thoughtcrime/securesms/loki/activities/SettingsActivity.kt:77]: General convention is to follow the indentation style the rest of the file uses

File: res/values/colors.xml [hunk 3]
```diff
@@ -107,4 +107,11 @@
     <color name="default_background_start">#121212</color>
     <color name="default_background_end">#171717</color>
 
+    <array name="user_pic_placeholder_primary">
```
Comment (nielsandriesse) [inline res/values/colors.xml:110]: Let's put this under the `<!-- Session -->` mark

File: res/values/colors.xml [hunk 4]
```diff
@@ -107,4 +107,11 @@
...
+        <item>#2bca81</item>
```
Comment (nielsandriesse) [inline res/values/colors.xml:111]: Let's make these colors match the ones in the design

File: res/drawable/profile_picture_view_small_background.xml [hunk 5]
```diff
@@ -6,6 +6,4 @@
     <solid  android:color="@color/unimportant" />
 
     <corners android:radius="18dp" />
-
-    <stroke android:width="@dimen/profile_picture_border_thickness" android:color="@color/border" />
```
Comment (nielsandriesse) [inline res/drawable/profile_picture_view_small_background.xml:10]: The placeholders in the design actually have borders

File: res/layout/view_profile_picture.xml [hunk 6]
```diff
@@ -19,13 +19,9 @@
                 android:id="@+id/doubleModeImageView1"
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
+                android:alpha="0.65"
```
Comment (nielsandriesse) [inline res/layout/view_profile_picture.xml:22]: I think I like this, but let's first implement the design we have and then think about changing it

File: src/org/thoughtcrime/securesms/loki/todo/AvatarPlaceholderGenerator.kt [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            !TextUtils.isEmpty(hashString) -> extractLabel(hashString)
+            else -> EMPTY_LABEL
+        }
+
+        val bitmap = Bitmap.createBitmap(pixelSize, pixelSize, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+
+        // Draw background/frame
+        val paint = Paint(Paint.ANTI_ALIAS_FLAG)
+        paint.shader = LinearGradient(0f, 0f, 0f, pixelSize.toFloat(),
```
Comment (nielsandriesse) [inline src/org/thoughtcrime/securesms/loki/todo/AvatarPlaceholderGenerator.kt:44]: Again, let's first get the design going cross-platform before changing it :)

File: src/org/thoughtcrime/securesms/notifications/SingleRecipientNotificationBuilder.java (hunk 1 above)
Comment (metaphore) [reply src/org/thoughtcrime/securesms/notifications/SingleRecipientNotificationBuilder.java:323]: Whoops, looks like I forgot about that one.

File: src/org/thoughtcrime/securesms/loki/activities/SettingsActivity.kt (hunk 2 above)
Comment (metaphore) [reply src/org/thoughtcrime/securesms/loki/activities/SettingsActivity.kt:77]: Sorry, you mean line breaks or the actual indentation (tabs/spaces)?

File: res/values/colors.xml (hunk 4 above)
Comment (metaphore) [reply res/values/colors.xml:111]: You talking about the color array for the old avatar drawable or the design document?

File: res/drawable/profile_picture_view_small_background.xml (hunk 5 above)
Comment (metaphore) [reply res/drawable/profile_picture_view_small_background.xml:10]: Yep, I took the liberty to remove it to show the difference. Most of the modern apps (messengers at least) that use circular avatars don't have borders and it looks cleaner IMHO. Please let me know if think we should stick to the old design, no problem going back.

File: src/org/thoughtcrime/securesms/loki/todo/AvatarPlaceholderGenerator.kt (hunk 7 above)
Comment (metaphore) [reply src/org/thoughtcrime/securesms/loki/todo/AvatarPlaceholderGenerator.kt:44]: Easy, this fancy gradient doesn't change that much anyway :)

**Pull request merged**